.github/workflows/_build_js_bindings.yml

```diff
@@ -0,0 +1,252 @@
+name: JS Bindings CI
+env:
+  DEBUG: napi:*
+  APP_NAME: "chromadb-js-bindings"
+  MACOSX_DEPLOYMENT_TARGET: '10.13'
+permissions:
+  contents: write
+  id-token: write
+'on':
+  workflow_dispatch: {}
+  workflow_call: {}
+jobs:
+  build-macos:
+    name: Build macOS bindings
+    runs-on: macos-latest
+    defaults:
+      run:
+        working-directory: rust/js_bindings
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 9
+          run_install: false
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: pnpm
+          cache-dependency-path: rust/js_bindings/pnpm-lock.yaml
+      - name: Set up Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
```

Comment:
did using our rust setup action not work here? we should ideally use it not just for dry but also because it reads from `rust-toolchain.toml` as the source of truth for the toolchain version
it also uses sccache so you don't need to handle caching (not entirely convinced the cache action setup here is correct)